test/vinyl/util.lua

```diff
@@ -0,0 +1,15 @@
+-- Check memory0 diff is as expected after inserting/replacing
+-- tuples. `payload` is total size of the tuples. `count` is
+-- number of inserted tuples.
+--
+-- Memory overhead should not be more than 10% of usage and not more
+-- than 40 byte per tuple.
+local function check_add_level0(before, payload, count)
+    local overhead = math.max(40 * count, payload * 0.1)
+    local diff = box.stat.vinyl().memory.level0 - before
+    return (diff > payload) and (diff < payload + overhead)
+end
+
+return {
+    check_add_level0 = check_add_level0,
+}
```

Comment:
Let's use in-place approximate checks like diff > 10 and diff < 100 and drop this function for the sake of simplicity. All we want to do is that level0 increases by some sane value, no need to be precise.

Reply:
Dropped `util.lua` helper and its mention in `test/vinyl/suite.ini`. Now patch looks like this (only hunks related to the usage of `check_add_level0` are shown).


```diff
diff --git a/test/vinyl/quota.test.lua b/test/vinyl/quota.test.lua
index a2793a015..248322560 100644
--- a/test/vinyl/quota.test.lua
+++ b/test/vinyl/quota.test.lua
@@ -20,21 +20,22 @@ sec = space:create_index('sec', { parts = {2, 'unsigned'} })

 space:insert({1, 1})

-box.stat.vinyl().memory.level0
+mem = box.stat.vinyl().memory.level0

 space:insert({1, 1})

-box.stat.vinyl().memory.level0
+box.stat.vinyl().memory.level0 == mem

 space:update({1}, {{'!', 1, 100}}) -- try to modify the primary key

-box.stat.vinyl().memory.level0
+box.stat.vinyl().memory.level0 == mem

 space:insert({2, 2})
 space:insert({3, 3})
 space:insert({4, 4})

-box.stat.vinyl().memory.level0
+delta = box.stat.vinyl().memory.level0 - mem
+delta > 50 and delta < 200

 box.snapshot()

@@ -46,7 +47,8 @@ box.stat.vinyl().memory.level0

 _ = space:replace{1, 1, string.rep('a', 1024 * 1024 * 5)}

-box.stat.vinyl().memory.level0
+box.stat.vinyl().memory.level0 > 5000000
+box.stat.vinyl().memory.level0 < 6000000

 space:drop()
diff --git a/test/vinyl/stat.test.lua b/test/vinyl/stat.test.lua
index 83c595509..7d27362c9 100644
--- a/test/vinyl/stat.test.lua
+++ b/test/vinyl/stat.test.lua
@@ -221,9 +223,10 @@ gstat().scheduler.compaction_input == istat().disk.compaction.input.bytes
 gstat().scheduler.compaction_output == istat().disk.compaction.output.bytes

 -- use memory
-st = gstat()
+st = box.stat.vinyl().memory.level0
 put(1)
-stat_diff(gstat(), st, 'memory.level0')
+diff = box.stat.vinyl().memory.level0 - st
+diff > 1000 and diff < 1100

 -- use cache
 st = gstat()
```